modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -123,19 +152,21 @@ class Configuration(object):
         self.password = password
         """Password for HTTP basic authentication
         """
+{{#hasHttpSignatureMethods}}
         if signing_info is not None:
             signing_info.host = host
         self.signing_info = signing_info
         """The HTTP signing configuration
         """
+{{/hasHttpSignatureMethods}}
 {{#hasOAuthMethods}}
-        self.access_token = ""
+        self.access_token = None
```

Comment:
This changes the default behavior of auth_settings.
Before the default of empty string would have the auth_settings() return a dict containing:
```
{
  ...
  'value': 'Bearer ',
  ...
}
```
Whereas this change would omit that entry.
Because this is a breaking change with fallback (users tests could break)
We can either:
- target the 4.3.0 (breaking change with fallback)
- or revert the access_token back to empty string in this PR and release it on master
What do you want to do here?
Did you have any discussions with @wing328 about this?

Reply:
But I argue this is a bug, just like password should not be the empty string by default. If a client actually wants to use a bearer token or OAuth2 access token, it will have to set the "access_token" attribute to a non-empty value.

Reply:
> But I argue this is a bug, just like password should not be the empty string by default. If a client actually wants to use a bearer token or OAuth2 access token, it will have to set the "access_token" attribute to a non-empty value.

I think that's what we do in other languages (e.g. PHP, Ruby) as empty string could be a possible value (e.g. password in HTTP basic auth but of course it's against security best practice) so it should not be set as the default.

Reply:
> target the 4.3.0 (breaking change with fallback)

Current master is (4.3.0), which is scheduled by the end of this month (Feb)

Reply:
So can I mark this conversation resolved?

Reply:
Yes, feel free to mark this as resolved